src/transformers/models/opt/modeling_opt.py

```diff
@@ -696,32 +840,12 @@ def forward(
         if inputs_embeds is None:
             inputs_embeds = self.embed_tokens(input_ids)
 
-        batch_size, seq_length = input_shape
         past_key_values_length = past_key_values[0][0].shape[2] if past_key_values is not None else 0
-        # required mask seq length can be calculated via length of past
-        mask_seq_length = past_key_values_length + seq_length
 
+        causal_attention_mask, attention_mask = self._update_causal_mask(
+            inputs_embeds, input_shape, past_key_values_length, attention_mask, output_attentions
+        )
```

Comment:
Are we missing the handling of the cross-attention mask expansion (the encoder attention mask)? 

See for reference bart again:
https://github.com/huggingface/transformers/blob/78b2929c0554b79e0489b451ce4ece14d265ead2/src/transformers/models/bart/modeling_bart.py#L1286-L1321

I think we could just copy past it from bart directly, wdyt.

Reply:
k

Reply:
Thx for clarifying, wasn't aware of this. Still a bit weird that the attention implementation mostly follows bart then, but not that it matters.

Reply:
Yeah, it was also weird to me, there was a refactor I tried to do, but I was told to revert it so it would still stay similar 😢